File: typescript-node-website-snippet/hello.ts
// Verbatim snippet from website/bindings/typescript-node/index.html (hello.ts)
import { DomainParticipantFactory } from '@zerodds/node';

async function main() {
    const factory = DomainParticipantFactory.instance();
    const participant = factory.createParticipant(0);

    const topic = participant.createBytesTopic('Chatter');
    const publisher = participant.createPublisher();
    const subscriber = participant.createSubscriber();

    const writer = publisher.createBytesWriter(topic);
    const reader = subscriber.createBytesReader(topic);

    await writer.waitForMatchedSubscription(1, 5000);
    await reader.waitForMatchedPublication(1, 5000);

    writer.write(new TextEncoder().encode('hello'));
    await reader.waitForData(3000);

    for (const sample of reader.take()) {
        console.log('got:', new TextDecoder().decode(sample));
    }
}

main().catch(console.error);

File: python-hello-bytes/hello.py
import zerodds

factory = zerodds.DomainParticipantFactory.instance()
participant = factory.create_participant(0)

topic = participant.create_bytes_topic("Chatter")
publisher = participant.create_publisher()
subscriber = participant.create_subscriber()

writer = publisher.create_bytes_writer(topic)
reader = subscriber.create_bytes_reader(topic)

writer.wait_for_matched_subscription(1, timeout_secs=5.0)
reader.wait_for_matched_publication(1, timeout_secs=5.0)

writer.write(b"hello")
reader.wait_for_data(timeout_secs=3.0)
for payload in reader.take():
    print(f"got: {payload!r}")
